frontend/gist-frontend/src/Components/DashboardComponent.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from "react";
+import { Box, Button, Text, Grommet } from "grommet";
+import { User, Group } from "grommet-icons";
+import { Row } from "antd";
+import { theme } from "../Helpers/theme";
+
+export default class DashboardComponent extends Component {
+  state = {
+    isGroupOpen: false
+  };
+
+  toggleGroupSidebar = () => {
+    const { isGroupOpen } = this.state;
+    this.setState({
+      isGroupOpen: !isGroupOpen
+    });
+  };
+
+  getOpenGroupsComponent = () => {
+    return (
+      <Row
+        type="flex"
+        justify="center"
+        align="top"
+        style={{ gridTemplateColumns: "4vw 11vw 85vw", display: "grid" }}
+      >
+        <div
+          style={{
+            height: "100vh",
+            backgroundColor: "#2AF598",
+            width: 82,
+            boxShadow: "10px 10px 5px grey"
+          }}
+        >
+          <Button key={"user"} href="#" hoverIndicator>
+            <Box
+              pad={{ horizontal: "medium", vertical: "small" }}
+              style={{ paddingLeft: 28 }}
+            >
+              <User color="light-2" />
+            </Box>
+          </Button>
+          <Button
+            key={"group"}
+            href="#"
+            hoverIndicator
+            onClick={this.toggleGroupSidebar}
+          >
+            <Box
+              pad={{ horizontal: "medium", vertical: "small" }}
+              style={{ paddingLeft: 28 }}
+            >
+              <Group color="light-2" />
+            </Box>
+          </Button>
+        </div>
+        <div
+          style={{
+            height: "100vh",
+            backgroundColor: "#2AF598",
+            opacity: 0.9,
+            width: 212
+          }}
+        ></div>
+
+        <div>Main Content</div>
+      </Row>
+    );
+  };
+
+  getClosedGroupsComponent = () => {
+    return (
+      <Row
+        type="flex"
+        justify="center"
+        align="top"
+        style={{ gridTemplateColumns: "4vw  96vw", display: "grid" }}
+      >
+        <div
+          style={{
+            height: "100vh",
+            backgroundColor: "#2AF598",
+            width: 78
+          }}
+        >
+          <Button key={"user"} href="#" hoverIndicator>
+            <Box
+              pad={{ horizontal: "medium", vertical: "small" }}
+              style={{ paddingLeft: 28 }}
+            >
+              <User color="light-2" />
+            </Box>
+          </Button>
+          <Button
+            key={"group"}
+            href="#"
+            hoverIndicator
+            onClick={this.toggleGroupSidebar}
+          >
```

Comment:
I know you're still working on this, but there is a lot of repeated code here that I have to at least mention even though you probably realize it, probably the only thing I see that could be updated here. I've never really worked with ant's row feature so don't know anything about that but this all looks promising. Although, I am wondering if we should keep with the rounded edges like the login page for consistency. This is looking alot like bitbucket which is definitely the point, but should we try and keep with the overall style of the app?